test/fixtures/rules/std_rule_cases/L023.yml

```diff
@@ -1,6 +1,41 @@
 rule: L023
 
 test_fail_cte_no_space_after_as:
-  # Check fixing of single space rules
+  # Check fixing of single space rule when space is missing
   fail_str: WITH a AS(select 1) select * from a
   fix_str: WITH a AS (select 1) select * from a
+
+test_fail_multiple_spaces_after_as:
+  # Check fixing of single space rule on multiple spaces
+  fail_str: WITH a AS  (select 1) select * from a
+  fix_str: WITH a AS (select 1) select * from a
+
+test_fail_cte_newline_after_as:
+  # Check fixing of replacing newline with space
+  fail_str: |
+    WITH a AS
+    (
+      select 1
+    )
+    select * from a
+  fix_str: |
+    WITH a AS (
+      select 1
+    )
+    select * from a
+
+test_fail_cte_newline_and_spaces_after_as:
+  # Check stripping newlines and extra whitespace
+  fail_str: |
+    WITH a AS
+
+
+        (
+      select 1
+    )
+    select * from a
+  fix_str: |
```

Comment:
A general question about the `yaml_test_cases_test.py`.

IMHO it's confusing and possibly dangerous that even if the `fail_str` + `fix_str` comparison passes, it could be possible that `lint` would _not_ pass. I presume `fix` would also return non-zero exit code in such case. Would it make sense to also ensure here that there are no unfixable errors remaining?

For example, this yaml test passes at the moment:
```
test_fail_comment:
  # Check fixing of single space rule on when there's a comment before bracketed section
  fail_str: |
    WITH a AS -- comment
    (select 1) select * from a
  fix_str: |
    WITH a AS -- comment
    (select 1) select * from a
```

If I modify the `rules.py` to be like this:
```python
        # If a `fixed` value is provided then check it matches
        if test_case.fix_str:
            assert res == test_case.fix_str
            # MY NEW CODE BELOW:
            assert_rule_pass_in_sql(
                test_case.rule,
                test_case.fix_str,
                configs=test_case.configs,
            )
```
Then the test case fails (like I would expect) as follows:

```
Errors Found: [<SQLLintError: rule L023 pos:(1, 8), #fixes: 0, description: Single whitespace expected after 'AS' in 'WITH' clause.>]
```

Maybe there can be legitimate cases where the point of test case is to just check that fix doesn't touch the query (or that it fixes it to some extent but not fully). But IMHO it should be explicitly defined by the test cases then that fix is expected to (also) return an error instead of succeeding.

Reply:
I’d say that’s a bad test, in that the `fix_str` shouldn’t be a copy of the `fail_str`. It should either be omitted (so it fails, but there’s an acceptance the rule can’t fix that scenario - this is what would be used for the rules that are not fixable), or it should be different.

But would be good to enforce this so no bad test cases can be added and “silently fail” as you state. If your proposed extra code does that then I think that would be a good extra addition. Maybe as a separate PR to this one?

Reply:
> I’d say that’s a bad test, in that the fix_str shouldn’t be a copy of the fail_str.

True. Indeed I'm more concerned if there can be cases where some fixing does happen (so that `fix_str` & `fail_str` differ) but but the fixing is not enough to make the rule pass.

> Maybe as a separate PR to this one?

Sure thing!